### DISCUSSION TITLE
230703 백엔드 기술 스택 회의(17:45 ~ 18:45)

### DISCUSSION THREAD

**Discussion by seokhwan-an** [woowacourse-teams/2023-shook#12] [회의]
### 회의 안건
프로젝트에 이용할 기술 스택을 정한다.

**Comment by splitCoding** [comment]
# 기술 스택

## Spring Boot 3.1.X

### Spring이 아닌 Spring Boot를 사용하는 이유는 ?

- Spring을 사용할 경우 외부 라이브러리나 여러가지 dependency를 추가하고 설정하는 많은 리소스가 필요하다.
- 구성해야하는 여러가지 세부적인 라이브러리에 대한 레퍼런스를 추상화된 spring boot 라이브러리로 제공한다.

### 버전을 3.1.X 이상으로 가는 이유는 ?

- Spring 2.X, 3.0 모두 지원기간이 2023.11월 까지이다.
- 프로젝트를 우테코 기간이 아닌 이후에도 운영하기 위해서는 지원기간이 긴 최신 버전을 택해야 한다.

## JAVA 17

- Spring Boot 3.1.X 의 경우 JAVA 17 이상을 지원한다.
- 17 이상의 버젼 중 유일한 LTS 버젼이다.

## Spring Data JPA

### ORM 프레임워크 사용하는 이유는?

- DB 주도 설계와 객체 지향 설계 패러다임 사이의 간극을 줄임으로써 객체지향 설계 집중할 수 있다.
- 코드가 DB에 의존되기 때문에 변경사항이 모든 쿼리에 영향을 미칠 수 있다.
- 세부적인 구현 내용을 파악하기 위해서 쿼리를 보지 않아도 된다.
- 데이터베이스 벤더의 변경에 유연하다.

## Lombok

- 코드의 가독성이 올라간다.
- 변경사항에 따른 코드 변경에 소비되는 리소스가 줄어든다. ( ex. 필드에 따른 생성자 변경 )

## H2 DB

- 메모리 사용량이 적어 소규모 애플리케이션 및 테스트 용도로 적합하다.
- 가벼운 설정과 빠른 구동 속도로 사용에 용이하다.
    - 인메모리로 사용하면 별도의 설치가 필요없다.
